### PR TITLE
mediawiki: introduce web pod autoscaler

### PR DESCRIPTION
**Pull request by tarrow**
This commit introduces a pod autoscaler for the mediawiki web pod. We have progressively increased this from 1 to 3 replicas over time so being able to have this dynamically scale would give us some headroom to hopefully avoid future issues around cpu exhaustion.

Enabling this is made configurable using a similar pattern of values to ingress-nginx[1]. Leaving it configurable keeps the possibility to not scale locally as well as turn it off if we find the autoscaler to be problematic.

80% cpu was chosen as it is the default for horizontal pod autoscaling.

[1] https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/values.yaml#L393

### CODE REVIEW

File: charts/mediawiki/templates/hpa-web.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  name: mediawiki-139-app-web
+  namespace: default
+spec:
+  maxReplicas: {{ .Values.web.autoscaling.maxReplicas }}
+  minReplicas: {{ .Values.web.autoscaling.minReplicas }}
+  metrics:
+  - resource:
+      name: cpu
+      target:
+        averageUtilization: 80
```
Comment (m90) [inline charts/mediawiki/templates/hpa-web.yaml:14]: Question: where is this number coming from? Just a guess or is it backed by anything? I'm not opposed to value, but would like to understand better where it's coming from.

Also: would it make sense to have it defined in `values`?

**Comment by m90** [comment]
General question: seeing the ingress nginx also scale on memory exhaustion, would it make sense to do the same here as well while we're at it?

File: charts/mediawiki/templates/hpa-web.yaml (hunk 1 above)
Comment (tarrow) [reply charts/mediawiki/templates/hpa-web.yaml:14]: I was a little vague in the PR
> 80% cpu was chosen as it is the default for horizontal pod autoscaling.

This is the default kubernetes has internally for doing this autoscaling see: [this bit of the docs](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2/#:~:text=If%20not%20set%2C%20the%20default%20metric%20will%20be%20set%20to%2080%25%20average%20CPU%20utilization).
.

It was chosen pretty randomly but I think given our issues were seeing cpu totally saturate this should be sufficient. I currently think we won't need to adjust this and therefore would suggest we make it a variable the first time we need to.

**Comment by tarrow** [comment]
> General question: seeing the ingress nginx also scale on memory exhaustion, would it make sense to do the same here as well while we're at it?

I think if we see more OOM on nginx this could certainly be a sensible thing to do; my expectation (as happened in this ticket) is that the memory exhaustion is because there becomes a huge backlog of connections (because MW is being so slow). I'd probably wait for seeing OOM in the ingress controller without cpu exhausted MW pods to do this.

**Review by m90 — APPROVED** [review]

**Pull request merged**